waspc/test/Analyzer/ParserTest.hs

```diff
@@ -5,319 +5,337 @@ import Data.Either (isLeft)
 import Test.Tasty.Hspec
 import Wasp.Analyzer.Parser
 
-spec_Parser :: Spec
```

Comment:
Git got confused with this file's diff

All I did was:
- add tests for `isValidWaspIdentifier`
- add tests for `parseExpression`
- removed the `describe` from `parseStatements` tests (because the output looked incorrect, I can explain if necessary)
- some renaming (`parse`  to `parseStatements`)